src/components/CourseCard/BaseCourseCard.jsx

```diff
@@ -3,53 +3,55 @@ import PropTypes from 'prop-types';
 import classNames from 'classnames';
 import moment from 'moment';
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
-
-import './CourseCard.scss';
+import { StatusAlert } from '@edx/paragon';
 
 import EmailSettingsModal from './EmailSettingsModal';
 
+import './CourseCard.scss';
+
 class BaseCourseCard extends Component {
   state = {
     modals: {
-      email_settings: null,
+      emailSettings: null,
     },
+    hasEmailsEnabled: this.props.hasEmailSettings || false,
+    hasNewEmailSettings: false,
   };
 
   setModalState = ({ key, options }) => {
-    this.setState({
+    this.setState(state => ({
       modals: {
-        ...this.state.modals,
+        ...state.modals,
         [key]: options,
       },
-    });
+    }));
+  };
+
+  handleEmailSettingsModalOnClose = (hasEmailsEnabled) => {
```

Comment:
This function is called when the `Modal` closes. If the `onClose` prop is manually called from within `EmailSettingsModal`, it will pass the `checked` state of the email setting checkbox.

Then, it will update the `BaseCourseCard` state to reflect the updated email settings and render the success status alert (through `hasNewEmailSettings` being truthy).